contracts/interface/IChainlinkPriceFeedV3.sol

```diff
@@ -0,0 +1,51 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.7.6;
+
+interface IChainlinkPriceFeedV3Event {
+    /// @param NotFreezed default state: Chainlink is working as expected
+    /// @param NoResponse fails to call Chainlink
+    /// @param InvalidTimestamp no timestamp or it’s invalid, either outdated or in the future
+    /// @param AnswerIsOutlier if the answer deviates more than _maxOutlierDeviationRatio
+    enum FreezedReason {
+        NotFreezed,
+        NoResponse,
+        IncorrectDecimals,
+        NoRoundId,
+        InvalidTimestamp,
+        NonPositiveAnswer,
+        AnswerIsOutlier
+    }
+
+    event ChainlinkPriceUpdated(uint256 price, uint256 timestamp, FreezedReason freezedReason);
+}
+
+interface IChainlinkPriceFeedV3 is IChainlinkPriceFeedV3Event {
+    struct ChainlinkResponse {
+        uint80 roundId;
+        int256 answer;
+        uint256 updatedAt;
+        bool success;
+        uint8 decimals;
+    }
+
+    /// @param interval twap interval
+    ///        when 0, cache price only, without twap; else, cache price & twap
+    /// @dev this is the non-view version of cacheTwap() without return value
+    function cacheTwap(uint256 interval) external;
+
+    function getLastValidPrice() external view returns (uint256);
+
+    function getLastValidTimestamp() external view returns (uint256);
+
+    /// @param interval twap interval
+    /// @dev this is the view version of cacheTwap()
+    function getCachedTwap(uint256 interval) external view returns (uint256);
+
+    function isTimedOut() external view returns (bool);
+
+    function getFreezedReason() external view returns (FreezedReason);
```

Comment:
This is for monitoring on BE s.t. we can know what's wrong!

Reply:
just a nice to have feedback:
i'd always try to minimize the scope of the contract if possible, since no other contract need this freezedReason, and ChainlinkPriceFeedV3 only need to know 2 reason: if it's NotFreezed or AnswerIsOutlier ( from _cachePrice() & _getCachePrice() ), so the rest are not relevant to the ChainlinkPriceFeedV3. 

but we can still do the same query from sth like a ChainlinkPriceFeedV3Monitor contract, to have a getFreezedReason(chainlinkV3PriceFeedAddr) and move those logic to that monitor contract. though in this case we need to expose _maxOutlierDeviationRatio ( then this suddenly remind me that it seems sth we should expose )

Reply:
Note: can take the viewer contract into consideration in earlier phase s.t. main contract can be cleaner!